azurerm/cluster/output.tf

```diff
@@ -0,0 +1,6 @@
+output "cluster" {
```

Comment:
I like the idea of these outputs. We should unify this for all three provider implementations. This also seems related to #136

Reply:
Outputting the whole cluster that is "hidden" on purpose in an internal module seems like a bit too much though. 

On the other hand this is definitely better then no outputs.

Reply:
Certainly something to discuss. Not exposing the cluster module like this, would require finding common denominators between the different providers. This has the risk of playing catch-up with the provider implementation.

Main purpose of the `_modules/*` as private is that it allows breaking changes to the input variables, while providing backwards compatibility for the public `cluster` and now `cluster-local` module.

Reply:
To make these changes work, there is no need to expose the entire cluster. The two keys needed are `node_resource_group` and `aks_nsg_id` which was added in this PR.

However, if we were to move the custom vnet into _modules/aks, as I mentioned in another comment thread, neither of these will be needed, as they will be available from the local context.

Reply:
In the latest iteration, which now creates a separate vnet to support CNI networking configuration, the cluster output has been removed, but an output containing the vnet has been added.

I also removed the `current_configuration` output. As discussed above, it would be generally useful for integrating Kubestack into wider Terraform project structures, but it should be added uniformly across all cloud providers, and would be more appropriate in a separate PR.